cloud.foundry.cli/src/main/java/cloud/foundry/cli/logic/apply/ServiceRequestsPlanner.java

```diff
@@ -15,157 +13,104 @@
-     */
-    @Override
-    public void visitObjectValueChanged(CfObjectValueChanged objectValueChanged) {
-        if (this.getRequests().size() == 0) {
-            Object affectedObject = objectValueChanged.getAffectedObject();
-            if (affectedObject instanceof ServiceBean) {
-                try {
-                    addUpdateServiceRequest((ServiceBean) affectedObject);
-                } catch (NullPointerException e) {
-                    throw new ApplyException(e);
-                }
-            } else {
-                throw new IllegalArgumentException("Only changes of services are permitted.");
-            }
-        }
-    }
+        if (hasNewObject(changes)) {
+            log.debug("Add create service request for service: " + serviceName);
 
-    /**
-     * Creates requests for Update Service.
-     *
-     * @param affectedObject ServiceBean
-     */
-    private void addUpdateServiceRequest(ServiceBean affectedObject) {
-        this.addRequest(this.servicesOperations.update(serviceName, affectedObject));
-    }
+            ServiceBean bean = (ServiceBean) getChange(changes, change -> change instanceof CfNewObject)
+                .get()
+                .getAffectedObject();
 
-    /**
-     * Creates the requests for CfRemovedObject.
-     *
-     * @param removedObject the CfRemovedObject to be visited
-     */
-    @Override
-    public void visitRemovedObject(CfRemovedObject removedObject) {
-        checkArgument(this.getRequests().size() == 0,
-                "There may not be any other requests for that service when adding a remove request.");
+            return Flux.merge(this.servicesOperations.create(this.serviceName, bean));
+        } else if (hasRemovedObject(changes)) {
+            log.debug("Add remove service request for service: " + serviceName);
 
-        if (!(removedObject.getAffectedObject() instanceof ServiceBean)) {
-            throw new IllegalArgumentException("Only changes of services are permitted.");
-        }
+            return Flux.merge(this.servicesOperations.remove(serviceName));
+        } else if (hasContainerChange(changes) || hasObjectValueChanged(changes)) {
+            log.debug("Add update service request for service: " + serviceName);
+            
+            ServiceBean bean = (ServiceBean) getChange(changes, 
+                 change -> change instanceof CfContainerChange || change instanceof CfObjectValueChanged)
+                .get()
+                .getAffectedObject();
 
-        try {
-            log.info("Requesting removal of service", serviceName);
-            this.addRequest(servicesOperations.remove(serviceName));
-        } catch (UpdateException | NullPointerException e) {
-            throw new ApplyException(e);
+            return Flux.merge(this.servicesOperations.update(serviceName, bean));
         }
+
+        return Flux.merge(requests);
```

Comment:
You can just write `return Flux.empty()`. On the other hand this should never be reached and may indicate a bug. So throwing an Exception is also possible.